ledger-core/virtual/integration/deduplication_test.go

```diff
@@ -318,3 +321,525 @@ func TestDeduplication_SecondCallOfMethodAfterExecution(t *testing.T) {
 
 	mc.Finish()
 }
+
+// test deduplication of method calls using VFindCallRequest
+// and previous virtual executor
+
+// we send single VCallRequest with method call and pulse of outgoing
+// less than the current. Object may have pending execution that is either
+// the request or some other one. Pending can be confirmed or not, finished
+// or not.
+
+// Depending on options we expect execution to happen or not and call result
+// message to be sent or not.
+
+type deduplicateMethodUsingPrevVETestInfo struct {
+	name    string
+
+	// pendings
+	pending             bool
+	confirmPending      bool
+	pendingIsTheRequest bool
+	finishPending       bool
+
+	// lookup
+	expectFindRequestMessage bool // is VFindCallRequest expected
+	findRequestStatus        payload.VFindCallResponse_CallState
+	findRequestHasResult     bool
+
+	// actions
+	expectExecution     bool // should request be executed
+	expectResultMessage bool // is VCallResult expected
+}
+
+func TestDeduplication_MethodUsingPrevVE(t *testing.T) {
+	t.Log("C5097")
+	t.Skip("https://insolar.atlassian.net/browse/PLAT-390")
+
+	table := []deduplicateMethodUsingPrevVETestInfo{
+		{
+			name: "no pendings, findCall message, missing call",
+
+			expectFindRequestMessage: true,
+			findRequestStatus:        payload.MissingCall,
+
+			expectExecution:     true,
+			expectResultMessage: true,
+		},
+		{
+			name: "no pendings, findCall message, unknown call",
+
+			expectFindRequestMessage: true,
+			findRequestStatus:        payload.UnknownCall,
+
+			expectExecution:     true,
+			expectResultMessage: true,
+		},
+		{
+			name: "no pendings, findCall message, found call, has result",
+
+			expectFindRequestMessage: true,
+			findRequestStatus:        payload.FoundCall,
+			findRequestHasResult:     true,
+
+			expectResultMessage: true,
+		},
+
```

Comment:
there must be test "no pendings, findCall message, found call, no result",

Reply:
I don't think so. I believe that it's situation that shouldn't happen in release 0. If previous executor knows about request then executor either produced result or request is in pending executions, but we don't have pendings, so it has result. In future releases such case may happen once we start "forgetting" results.

Reply:
I would rather see panic Impossible if we got "found without result", pendings table is complete and request is not in the table, but I didn't write test for that.

Reply:
I can make panic in my code and add a test for that case